flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServicesTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.RestOptions;
+import org.apache.flink.runtime.blob.BlobServer;
+import org.apache.flink.runtime.blob.VoidBlobStore;
+import org.apache.flink.runtime.util.TestingFatalErrorHandler;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests auxiliary shared services created by {@link JobManagerSharedServices} and used by the
+ * {@link JobMaster}.
+ */
+public class JobManagerSharedServicesTest extends TestLogger {
+
+    private static final int NUM_FUTURE_THREADS = 128;
+    private static final int NUM_IO_THREADS = 256;
+
+    @Test
+    public void testBasicMapping() throws Exception {
+        Configuration config = new Configuration();
+        config.setInteger(RestOptions.JOBMANAGER_FUTURE_THREADS, NUM_FUTURE_THREADS);
+        config.setInteger(RestOptions.JOBMANAGER_IO_THREADS, NUM_IO_THREADS);
+
+        final JobManagerSharedServices jobManagerSharedServices =
+                JobManagerSharedServices.fromConfiguration(
+                        config,
+                        new BlobServer(config, new VoidBlobStore()),
+                        new TestingFatalErrorHandler());
+        assertEquals(
+                NUM_FUTURE_THREADS,
+                ((ThreadPoolExecutor) jobManagerSharedServices.getFutureExecutor())
+                        .getCorePoolSize());
+        assertEquals(
+                NUM_IO_THREADS,
+                ((ThreadPoolExecutor) jobManagerSharedServices.getIoExecutor()).getCorePoolSize());
```

Comment:
This test is a bit brittle because it relies on `jobManagerSharedServices.getFutureExecutor()` being a `ThreadPoolExecutor`. If the implementation changes, then this test will break.

Reply:
One way to test the behaviour could be to submit `Runnables` that count down a `CountDownLatch` and then block on another latch. If you have two latches `a = new CountDownLatch(poolSize)` and `b = new CountDownLatch(poolSize + 1)` and then submit `poolSize + 1` runnables, you should see that `a` fires and `b` not.